src/Symfony/Component/Validator/Constraints/ContainsWordValidator.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+class ContainsWordValidator extends AbstractStringContainsValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function doValidate(string $value, iterable $texts, AbstractStringContains $constraint): void
+    {
+        foreach ($texts as $text) {
+            $pattern = sprintf('/\\b%s\\b/%s', preg_quote($text, '/'), $constraint->caseSensitive ? '' : 'i');
```

Comment:
IMHO this one doesnt add enough value over RegexValidator :/

Reply:
then again, a complete set of rich constraints (including StartsWith, EndWith) is nice to have. No blocker for me either way.

Reply:
> IMHO this one doesnt add enough value over RegexValidator :/

The point is UX / code readability

```
@Assert\NotContainsWord({"some", "bad", "words"})
```

vs

```
@Assert\Regex("/^(?!.*(some|bad|words)).*$/", match=false)
```

>  then again, a complete set of rich constraints (including StartsWith, EndWith) is nice to have. No blocker for me either way.

:+1: i'll add them